src/regressionmodel.jl

```diff
@@ -34,19 +34,25 @@ Return the mean of the response.
 function meanresponse end
 
 """
-    modelmatrix(model::RegressionModel)
+    modelmatrix(model::RegressionModel; weighted::Bool=false)
 
-Return the model matrix (a.k.a. the design matrix).
+Return the model matrix (a.k.a. the design matrix) or, if `weighted=true` the weighted 
+model matrix, i.e. `X' * sqrt.(W)`, where `X` is the model matrix and
+`W` is the diagonal matrix whose elements are the model weights. 
 """
-function modelmatrix end
+function modelmatrix(model::RegressionModel; weighted::Bool=false) end
 
 """
-    crossmodelmatrix(model::RegressionModel)
+    crossmodelmatrix(model::RegressionModel; weighted::Bool=false)
 
-Return `X'X` where `X` is the model matrix of `model`.
+Return `X'X` where `X` is the model matrix of `model` or, if `weighted=true`, `X'WX`, 
+where `W` is the diagonal matrix whose elements are the model weights. 
```

Comment:
Can we define weights?

Reply:
We could add a link to the `weights(::StatisticalModel)` method. Indeed there can be confusion between prior weights and working weights (though these terms can also confuse casual users).

Reply:
How exactly do I add a link to `weights(::StatisticalModel)`? Is there a way to link docs from different packages?

Reply:
It's in the same package so I think something like `[model weights](@ref weights(::StatisticalModel))` should work. Better test it though by building the StatsBase docs (`julia docs/make.jl`) using the updated StatsAPI.